src/hotspot/share/prims/scopedMemoryAccess.cpp

```diff
@@ -138,17 +159,43 @@ class CloseScopedMemoryClosure : public HandshakeClosure {
 };
 
 /*
- * This function issues a global handshake operation with all
- * Java threads. This is useful for implementing asymmetric
- * dekker synchronization schemes, where expensive synchronization
- * in performance sensitive common paths, may be shifted to
- * a less common slow path instead.
- * Top frames containg obj will be deoptimized.
+ * This functin performs a thread-local handshake against all threads running at the time
+ * the given scope (deopt) was closed. If the handshake closure finds that a thread has
+ * safepointed inside a scoped method (that is, a method inside the ScopedMemoryAccess class
```

Comment:
Do you really mean "safepointed" here, or do you just mean the point where the handshake is processed?